File: control-flow/Q104/main.go
package main

import "fmt"

func main() {
	var ch rune
	for {
		fmt.Print("Enter a char: ")
		fmt.Scanf("%c\n", &ch)
		switch ch {
		case 'b', 'B':
			fmt.Println("You selected Lady")
		case 'd', 'D':
			fmt.Println("You selected Miss")
		case 'p', 'P':
			fmt.Println("You selected Professor")
		case 'a', 'A':
			fmt.Println("You selected Mr")
		case 'j', 'J':
			fmt.Println("You selected Excellency")
		case 'm', 'M':
			fmt.Println("You selected Wife")
		default:
			return
		}
	}
}

File: control-flow/Q104/main.rs
use std::io::{self, Write};

fn main() {
    let mut input = String::new();

    loop {
        print!("Enter a char: ");
        io::stdout().flush().expect("Failed to flush stdout");

        input.clear();
        if io::stdin().read_line(&mut input).is_err() {
            return;
        }

        let ch = match input.chars().find(|c| *c != '\n' && *c != '\r') {
            Some(c) => c,
            None => return,
        };

        match ch {
            'b' | 'B' => println!("You selected Lady"),
            'd' | 'D' => println!("You selected Miss"),
            'p' | 'P' => println!("You selected Professor"),
            'a' | 'A' => println!("You selected Mr"),
            'j' | 'J' => println!("You selected Excellency"),
            'm' | 'M' => println!("You selected Wife"),
            _ => return,
        }
    }
}